cardinal/state_test.go

```diff
@@ -170,7 +163,7 @@ func TestCanRecoverArchetypeInformationAfterLoad(t *testing.T) {
 
 	// Create a brand new engine, but use the original redis store. We should be able to load
 	// the game state from the redis store (including archetype indices).
-	tf2 := testutils.NewTestFixture(t, redisStore)
+	tf2 := testutils.NewTestFixture(t, nil)
```

Comment:
Shoot, this test has a bug in it. The goal was to make sure archetypes are remembered across world instances as long as the same DB is used. This line here should be `tf2 := testutils.NewTestFixture(t, tf1.Redis)` (the tf3 line below should also be updated.)

But this test is still passing because world1 is being used to load archetypes.

Please fix this test. Lines 177, 180, and 183 should be `... := world2.GameStateManager()...`

And similar changes on lines 199, 202, and 204: `... := world3.GameStateManager()...`

With those changes, this test should now properly fail, which can be fixed by using tf1.Redis instead of nil.